pkgs/compiler-lib/compiler/commands/test.rkt

```diff
@@ -1092,8 +1093,14 @@
  [("--table" "-t")
   "Print a summary table"
   (set! table? #t)]
+ #:once-each
+ [("--output" "-o") filename
+  "Save stdout and stderr to file"
+  (set! default-output-port (open-output-file (string->path filename) #:exists 'truncate))
```

Comment:
I think you need to close the port at some point, right?

It might be worth it to switch `open-output-file` to `with-output-to-file`. That is, make the code below (`(let () ...)`) a thunk, and pass this thunk to an invoker. When `--output` is not set, the invoker will invoke the thunk right away. However, if `--output` is set, then the invoker will supply the thunk to `with-output-to-file`.

Reply:
Good idea, the port should be closed.

I assume that if the user specified a --output when invoking raco test, the stderr also needs to be pipelined into the same file, in this case, `with-output-to-file` doesn't work as far as I know.

Reply:
You can `parameterize` the `current-error-port` to be `current-output-port`.

Reply:
`with-output-to-file` doesn't work here, I implemented it with `call-with-output-file*`.